workitem/expression_compiler_blackbox_test.go

```diff
@@ -23,26 +26,67 @@ func TestField(t *testing.T) {
-		), `((`+workitem.Column("iter", "name")+` = ?) OR (`+workitem.Column("ar", "name")+` = ?))`, []interface{}{"abcd", "xyz"}, []string{"iteration", "area"})
-
-		expect(t, c.Or(
-			c.Equals(c.Field("iteration.name"), c.Literal("abcd")),
-			c.Equals(c.Field("iteration.created_at"), c.Literal("123")),
-		), `((`+workitem.Column("iter", "name")+` = ?) OR (`+workitem.Column("iter", "created_at")+` = ?))`, []interface{}{"abcd", "123"}, []string{"iteration"})
+		t.Run("iteration", func(t *testing.T) {
+			j := *defJoins["iteration"]
+			j.Active = true
+			j.HandledFields = []string{"name"}
+			expect(t, c.Equals(c.Field("iteration.name"), c.Literal("abcd")), `(`+workitem.Column("iter", "name")+` = ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+		})
+		t.Run("area", func(t *testing.T) {
+			j := *defJoins["area"]
+			j.Active = true
+			j.HandledFields = []string{"name"}
+			expect(t, c.Equals(c.Field("area.name"), c.Literal("abcd")), `(`+workitem.Column("ar", "name")+` = ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+		})
+		t.Run("codebase", func(t *testing.T) {
+			j := *defJoins["codebase"]
+			j.Active = true
+			j.HandledFields = []string{"url"}
+			expect(t, c.Equals(c.Field("codebase.url"), c.Literal("abcd")), `(`+workitem.Column("cb", "url")+` = ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+		})
+		t.Run("work item type", func(t *testing.T) {
+			j := *defJoins["work_item_type"]
+			j.Active = true
+			j.HandledFields = []string{"name"}
+			expect(t, c.Equals(c.Field("wit.name"), c.Literal("abcd")), `(`+workitem.Column("wit", "name")+` = ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+			expect(t, c.Equals(c.Field("work_item_type.name"), c.Literal("abcd")), `(`+workitem.Column("wit", "name")+` = ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+			expect(t, c.Equals(c.Field("type.name"), c.Literal("abcd")), `(`+workitem.Column("wit", "name")+` = ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+		})
+		t.Run("space", func(t *testing.T) {
+			j := *defJoins["space"]
+			j.Active = true
+			j.HandledFields = []string{"name"}
+			expect(t, c.Equals(c.Field("space.name"), c.Literal("abcd")), `(`+workitem.Column("space", "name")+` = ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+		})
+		t.Run("creator", func(t *testing.T) {
+			j := *defJoins["creator"]
+			j.Active = true
+			j.HandledFields = []string{"full_name"}
+			expect(t, c.Equals(c.Field("creator.full_name"), c.Literal("abcd")), `(`+workitem.Column("creator", "full_name")+` = ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+			expect(t, c.Equals(c.Field("author.full_name"), c.Literal("abcd")), `(`+workitem.Column("creator", "full_name")+` = ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+			expect(t, c.Not(c.Field("author.full_name"), c.Literal("abcd")), `(`+workitem.Column("creator", "full_name")+` != ?)`, []interface{}{"abcd"}, []*workitem.TableJoin{&j})
+		})
+		t.Run("iteration + area", func(t *testing.T) {
```

Comment:
just to be sure, this query can return results across multiple spaces, right?

Reply:
Theoretically, yes because it only limits by the iteration/area name which could be the same in multiple spaces.